frontend/src/components/editor/renderers/vertical-layout/vertical-layout.tsx

```diff
@@ -76,35 +76,37 @@ const VerticalLayoutRenderer: React.FC<VerticalLayoutProps> = ({
 
   return (
     <VerticalLayoutWrapper invisible={invisible} appConfig={appConfig}>
-      {cells.map((cell) => (
-        <VerticalCell
-          key={cell.id}
-          cellId={cell.id}
-          output={cell.output}
-          consoleOutputs={cell.consoleOutputs}
-          status={cell.status}
-          code={cell.code}
-          config={cell.config}
-          stopped={cell.stopped}
-          showCode={showCode && canShowCode}
-          errored={cell.errored}
-          mode={mode}
-          runStartTimestamp={cell.runStartTimestamp}
-          interrupted={cell.interrupted}
-          staleInputs={cell.staleInputs}
-          name={cell.name}
-          kiosk={kioskMode}
-        />
-      ))}
-      {mode === "read" && (
-        <ActionButtons
-          canShowCode={canShowCode}
-          showCode={showCode}
-          onToggleShowCode={() => setShowCode((v) => !v)}
-        />
-      )}
-      {/* Shown in read and present */}
-      <FloatingOutline />
+      <div className={showCode ? "flex flex-col gap-5" : undefined}>
```

Comment:
Can you just pass this class to the vertical layout? Avoid a bunch of nesting

Reply:
That doesn't quite work because vertical layout has its own nesting. Plus I realized we don't want to put everything (TOC, action buttons) in the flex, just the cell array.

Updated to only wrap in div when needed. So now the DOM is analogous to the `CellArray` component, which also wraps cells in a flexed div under a `VerticalLayoutWrapper`.